### ISSUE TITLE
Lint/Debugger not detecting issues inside block

### ISSUE THREAD

**Issue by Drowze** [rubocop/rubocop#12244]
When e.g.: `binding.pry` is inside a block, `Lint/Debugger` is failing to flag the issue.

--------

## Expected behavior

I would expect `Lint/Debugger` to always flag a code with `binding.pry`, regardless of where it is.

## Actual behavior

Lint/Debugger is not detecting `binding.pry` calls when they happen inside block.

## Steps to reproduce the problem

Given:
```ruby
def uri_escaped_path(uri)
  uri.path = uri.path.gsub(%r{[^/]+}) do |segment|
    binding.pry
    CGI.escape(segment.to_s.encode("UTF-8")).gsub("+", "%20").gsub("%7E", "~")
  end
  uri
end
```

When I run `rubocop --only Lint/Debugger`, nothing is flagged.

See:
![Screenshot 2023-10-04 at 09 48 25](https://github.com/rubocop/rubocop/assets/9031589/859cce73-ad9d-4f97-85fb-3b0c21c5e66a)

## RuboCop version

```
$ rubocop -V
1.56.4 (using Parser 3.2.2.4, rubocop-ast 1.29.0, running on ruby 3.2.2) [x86_64-darwin22]
```

**Comment by Drowze** [comment]
Debugged a bit and I see this bug was introduced between 1.45.1 and 1.46.0.
